addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -79,11 +81,19 @@ options.registry.TableOfContent = options.Class.extend({
      */
     _generateNav: function (ev) {
         this.options.wysiwyg && this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
+        const navEl = this.$target[0].querySelector('.s_table_of_content_navbar');
         const headingsEls = this.$target.find(this.targetedElements).toArray()
             .filter(el => !el.closest('.o_snippet_desktop_invisible'));
         const areHeadingsEqual = this.oldHeadingsEls.length === headingsEls.length
             && this.oldHeadingsEls.every((el, i) => el.isEqualNode(headingsEls[i]));
-        if (areHeadingsEqual) {
+        const areVisibilityIdsMatching = headingsEls.every((headingEl) => {
+            const visibilityId = headingEl.closest('section').getAttribute('data-visibility-id');
+            const matchingLinkEl = navEl.querySelector(`a[href="#${headingEl.getAttribute('id')}"]`);
+            const matchingLinkVisibilityId = matchingLinkEl ? matchingLinkEl.getAttribute('data-visibility-id') : null;
+            // Check if visibilityId matches matchingLinkVisibilityId or both are null/undefined
```

Comment:
```suggestion
            // Check if visibilityId matches matchingLinkVisibilityId or both
            // are null/undefined
```
https://github.com/odoo/odoo/pull/83508#discussion_r796442104

Reply:
Done.